### PR TITLE
[BUG][API]task definition reference deleted resource files.

### LINKED ISSUE #11827: task definition reference deleted resource files
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

1、Create a process  definiiton（flow001)  and add a shell task. The shell task references two rosource files (file1.sql and file2.sql）
2、Make flow001 offline 
3、From resource center, delete file1.sql
4、 Make flow001 online again
5、 The flow001 can not be scheduled  normally

### What you expected to happen

After delete file1.sql, I expected the flow001 can be scheduled normally

### How to reproduce

1、Create a process  definiiton（flow001)  and add a shell task. The shell task references two rosource files (file1.sql and file2.sql）
2、Make flow001 offline 
3、From resource center, delete file1.sql
4、 Make flow001 online again
5、 The flow001 can not be scheduled 

### Anything else

The resouce center reupload file is a very useful function . I suggest this function can be added back again

### Version

3.0.0

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hstdream**
## Purpose of the pull request

close #11827

## Brief change log

1、Create a process definiiton（flow001) and add a shell task. The shell task references two rosource files (file1.sql and file2.sql）
2、Make flow001 offline
3、From resource center, delete file1.sql
4、 Make flow001 online again
5、 The flow001 can not be scheduled normally
## Verify this pull request
1、Create a process definiiton（flow001) and add a shell task. The shell task references two rosource files (file1.sql and file2.sql）
2、Make flow001 offline
3、From resource center, delete file1.sql
4、 Make flow001 online again
5、 The flow001 can not be scheduled normally

### CODE REVIEW

**Comment by hstdream** [comment]
version in 3.0.x.

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionMapper.xml [hunk 1]
```diff
@@ -172,7 +172,7 @@
         FROM t_ds_process_task_relation ptr
         join t_ds_process_definition pd
             on ptr.process_definition_code=pd.code and ptr.process_definition_version = pd.version
-                and ptr.project_code=pd.project_code and  pd.release_state = 1
+                and ptr.project_code=pd.project_code
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionMapper.xml:158]: But I found `processDefinitionMapper.listResources()` is just called by UT, please check it.

Comment (hstdream) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionMapper.xml:158]: This problem only exists in 3.0.x, the code in dev is different.

**Review by zhongjiajie — APPROVED** [review]
good catch, LGTM

**Comment by zhongjiajie** [comment]
oh, I found @ruanwenjun had comment at https://github.com/apache/dolphinscheduler/issues/11827#issuecomment-1299414607. I change my mind and close it, FYI @hstdream

**Pull request closed without merging**